pkg/function/client/client.go

```diff
@@ -179,24 +180,15 @@ outputLoop:
 // setConn function is used to populate the connection properties based
 // on multiprocessing TCP or UDS connection
 
-func regMultProcResolver() {
-	maxProcs := runtime.GOMAXPROCS(0)
-	numCpu := runtime.NumCPU()
-	if maxProcs < numCpu {
-		numCpu = maxProcs
-	}
-	val, present := os.LookupEnv("NUM_CPU_MULTIPROC")
-	if present {
-		num, err := strconv.Atoi(val)
-		if err != nil || num < 1 {
-			numCpu = maxProcs
-		} else {
-			numCpu = num
-		}
+func regMultProcResolver(svrInfo *info.ServerInfo) error {
```

Comment:
Before this change, if we can't find `CPU_LIMIT` or can't concert it to a number, we set the `numCpu` to `runtime.GOMAXPROCS(0)` and continue starting the client using TCP with no error.

With this change, if so, we throw and fail the client creation.

For servers with serverInfo {protocol: TCP} but no `CPU_LIMIT` metadata, this could be a backward-incompatible change. Is this as expected?

Reply:
there are default values exposed via env field

Reply:
I think they come from env field and get populated to the `metadata` from [here](https://github.com/numaproj/numaflow-python/pull/89/files#diff-a7e5474397ecc3f74a92417dfb2cc787227c949e645b2edd8c82e4d13b586a04R239) in Python Server.

But for Java and Golang, currently the ServerInfo metadata is an empty map.

Reply:
The info-server should be exposing the CPU_LIMIT metadata whenever we want to start a TCP server because we want to maintain consistency in the number of server and client instances in the multiproc server. 
 I think if we don't get this info that should indicate an issue with the info-server (ergo an issue in our platform) and thus not start the client.
We could log the error and continue with a default value (maybe 1) in this case. 
@vigith @whynowy What do you guys suggest?

Reply:
I think it's good for now since only python has the chance to start multiprocessing server, which means it won't be called for the reset.